component/module/string/string_test.go

```diff
@@ -18,96 +18,198 @@ import (
+	}`
+
+const exportModuleLoaderConfig = `
+	local.file "exporter" { filename = "%exp" }
+	
+	module.string "exporter" {
+		content = local.file.exporter.content
+	}`
+
+func TestModule(t *testing.T) {
+	tt := []struct {
+		name                  string
+		riverContent          string
+		argumentModuleContent string
+		exportModuleContent   string
+		expectedComponentId   string
+		expectedExports       []string
+		expectedErrorContains string
+	}{
+		{
+			name:                  "Export String",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig,
+			argumentModuleContent: argumentConfig,
+			exportModuleContent:   exportStringConfig,
+			expectedComponentId:   "module.string.importer",
+			expectedExports:       []string{"password1", "bob"},
+		},
+		{
+			name:                  "Export Component",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig,
+			argumentModuleContent: argumentConfig,
+			exportModuleContent:   exportStringConfig + exportComponentConfig,
+			expectedComponentId:   "module.string.exporter",
+			expectedExports:       []string{"username", "password", "dummy"},
+		},
+		{
+			name:         "Empty Content Allowed",
+			riverContent: `module.string "empty" { content = "" }`,
+		},
+		{
+			name:                  "Argument blocks not allowed in parent config",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig + argumentConfig,
+			argumentModuleContent: argumentConfig,
+			exportModuleContent:   exportStringConfig,
+			expectedErrorContains: "argument blocks only allowed inside a module",
+		},
+		{
+			name:                  "Export blocks not allowed in parent config",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig + exportStringConfig,
+			argumentModuleContent: argumentConfig,
+			exportModuleContent:   exportStringConfig,
+			expectedErrorContains: "export blocks only allowed inside a module",
+		},
+		{
+			name:                  "Logging blocks not allowed in module config",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig,
+			argumentModuleContent: argumentConfig + loggingConfig,
+			exportModuleContent:   exportStringConfig,
+			expectedErrorContains: "logging block not allowed inside a module",
+		},
+		{
+			name:                  "Tracing blocks not allowed in module config",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig,
+			argumentModuleContent: argumentConfig + tracingConfig,
+			exportModuleContent:   exportStringConfig,
+			expectedErrorContains: "tracing block not allowed inside a module",
+		},
+		{
+			name:                  "Argument not defined in module source",
+			riverContent:          argumentModuleLoaderConfig + exportModuleLoaderConfig,
+			argumentModuleContent: `argument "different_argument" {}`,
+			exportModuleContent:   exportStringConfig,
+			expectedErrorContains: "is not defined in the module",
```

Comment:
I removed the check on a specific argument name here because this is not deterministic for some reason. The test would flip between username and password in the error message.